qa/pull-tester/rpc-tests.py

```diff
@@ -266,7 +268,23 @@ def main():
         subprocess.check_call((config["environment"]["SRCDIR"] + '/qa/rpc-tests/' + test_list[0]).split() + ['-h'])
         sys.exit(0)
 
-    run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+    if (args.rpcgroup == -1) != (args.machines == -1):
+        print("ERROR: Please use both -m and -r options when using parallel rpc_groups.")
+        sys.exit(0)
+    if args.rpcgroup > args.machines:
+        print("ERROR: -r/--rpcgroup must be less than -m/--machines")
+        sys.exit(0)
+    if args.rpcgroup != -1 and args.machines != -1:
+        # Ceiling division using floor division, by inverting the world.
+        # https://stackoverflow.com/a/17511341
+        k = -(len(test_list) // -args.machines)
+        split_list = list(test_list[i*k:(i+1)*k] for i in range(args.machines))
+        tests_to_run = split_list[args.rpcgroup]
+    else:
+        tests_to_run = test_list
+    print(len(tests_to_run))
+    print(tests_to_run)
+    #run_tests(tests_to_run, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
```

Comment:
Why is this commented out?

Reply:
I'm testing all the cases atm sorry for typo. Will clean up in a bit